frontend/model/contracts/chatroom.js

```diff
@@ -307,25 +307,28 @@ sbp('chelonia/defineContract', {
     },
     'gi.contracts/chatroom/leave': {
       validate: objectOf({
-        username: optional(string), // coming from the gi.contracts/group/leaveChatRoom
-        member: string // username to be removed
+        username: string,
+        // NOTE: 'operator' is someone who works as main actor of this action.
+        //       It is used to select profile picture of message inside the ChatMain.vue.
+        //       'operator' always equals to 'username' unless 'operator' kicks 'username' in the chatroom.
+        operator: string
```

Comment:
Why isn't this using `meta.username` instead of `operator`?

If there is some strong reason to use `operator`, can you explain in the comment why `meta.username` isn't being used?

Reply:
If a user leaves group, he will leave all the chatrooms of the group as well.
If **u2** kicks the **u1** from the group, the `meta.username` is **u2**. But in the chatroom it should be `Left General` with the profile picture of **u1**. Also if **u2** created proposal to remove **u1** from the group and **u3** accepts that proposal, the `meta.username` would be **u3**.

Reply:
I still don't understand. :-\

`operator` doesn't seem to be used by `ChatMain` or anywhere else in the codebase.

To display u1's profile picture you use `username`.

So I still don't understand the purpose of `operator`. To know who kicked or removed the user from the group, you can use `meta.username`. Then if `meta.username !== username` you know that the user was removed by someone else.

Reply:
> `operator` doesn't seem to be used by ChatMain or anywhere else in the codebase.

It's used in the function `'gi.actions/group/leaveChatRoom'`.

> So I still don't understand the purpose of operator.

What I mean is that `meta.username` can't be always the profile picture of notification message that saying `Left General` or something like that. In the above examples, the profile pictures of **u2**, **u3** can't be used for the notification message of `Left General` for u1.

Reply:
> It's used in the function 'gi.actions/group/leaveChatRoom'.

That is where it originates from, and that is what sends it to `'gi.contracts/chatroom/leave'`, but it is not actually used.

> What I mean is that meta.username can't be always the profile picture of notification message that saying Left General or something like that

I was never suggesting that `meta.username` be used to to show the profile picture. I said  right here:

> To display u1's profile picture you use `username`.

So again, please delete `operator` since it's not used anywhere, and use `username` for displaying the profile picture.

Reply:
> To display u1's profile picture you use username.

Without the `operator`, we can not decide when we should use `data.username` and when we should use `meta.username` in the chatroom contract. It is decided in the `'gi.actions/group/leaveChatRoom'` function with the option parameter `leavingGroup`.

Reply:
OK, per Slack, please change `operator` to `showKickedBy`, but also please change the logic to also reflect this simplified understanding.

We show `Left general` notification with u1 picture when `showKickedBy` is `undefined` or `null`. And we show `u2 kicked u1` message with profile picture of u2 when `showKickedBy` is set to u2's username